reactive/ovn_chassis_charm_handlers.py

```diff
@@ -181,3 +181,54 @@ def provide_chassis_certificates_to_principal():
                             level=ch_core.hookenv.WARNING)
 
     reactive.clear_flag('ovn.certs.changed')
+
+
+@reactive.when_any('config.changed.ovs-exporter-channel',
+                   'snap.installed.prometheus-ovs-exporter')
+def reassess_exporter():
```

Comment:
If you look at the other functions in the reactive handler file, they always call into the charm class to perform actions, unless the action is to call into into interface code or to set/clear flags.

Since you actually need charm class variables to perform your work, I would lean towards that it would be more clean to just move this into the charm class?

The resulting method also needs a unit test.

Reply:
Thanks for the feedback - you are right.

Addressed in https://github.com/openstack-charmers/charm-layer-ovn/pull/74/commits/0df1407c7afabd9c339e963890374696c4205e53